lib/jetstream/pull_consumer/server.ex

```diff
@@ -5,34 +5,34 @@ defmodule Jetstream.PullConsumer.Server do
 
   use Connection
 
-  alias Jetstream.PullConsumer.ConnectionOptions
+  alias Jetstream.PullConsumer.{ConnectionOptions, Worker}
 
   defstruct [
     :connection_options,
-    :state,
-    :listening_topic,
+    :connection_pid,
+    :worker_init,
+    :worker_pid,
     :module,
-    :subscription_id,
     current_retry: 0
   ]
 
+  @impl Connection
   def init(%{module: module, init_arg: init_arg}) do
     _ = Process.put(:"$initial_call", {module, :init, 1})
 
     case module.init(init_arg) do
-      {:ok, state, connection_options} when is_list(connection_options) ->
+      {:ok, worker_init, connection_options} when is_list(connection_options) ->
         Process.flag(:trap_exit, true)
 
         connection_options = ConnectionOptions.validate!(connection_options)
 
-        gen_state = %__MODULE__{
+        state = %__MODULE__{
           connection_options: connection_options,
-          state: state,
-          listening_topic: new_listening_topic(connection_options),
+          worker_init: worker_init,
```

Comment:
I think this rename only makes sense if we are going to allow multiple parallel workers. This would potentially allow each worker to start from the same initial value, but then update that value as they run, but I agree that this doesn't make sense if the use-case here is to have a single worker.